src/vmm/src/vmm_config/mod.rs

```diff
@@ -97,38 +97,19 @@ impl TryInto<RateLimiter> for RateLimiterConfig {
 
 type Result<T> = std::result::Result<T, std::io::Error>;
 
-/// Structure `Writer` used for writing to a FIFO.
-pub struct Writer {
-    line_writer: io::LineWriter<File>,
+/// Create and open a FIFO for writing to it.
+/// In order to not block the instance if nobody is consuming the message that is flushed to the
+/// two pipes, we are opening it with `O_NONBLOCK` flag. In this case, writing to a pipe will
+/// start failing when reaching 64K of unconsumed content.
+fn open_fifo(fifo_path: &PathBuf) -> Result<File> {
+    OpenOptions::new()
+        .custom_flags(O_NONBLOCK)
+        .read(true)
+        .write(true)
+        .open(&fifo_path)
 }
 
-impl Writer {
-    /// Create and open a FIFO for writing to it.
-    /// In order to not block the instance if nobody is consuming the message that is flushed to the
-    /// two pipes, we are opening it with `O_NONBLOCK` flag. In this case, writing to a pipe will
-    /// start failing when reaching 64K of unconsumed content. Simultaneously,
-    /// the `missed_metrics_count` metric will get increased.
-    pub fn new(fifo_path: PathBuf) -> Result<Writer> {
-        OpenOptions::new()
-            .custom_flags(O_NONBLOCK)
-            .read(true)
-            .write(true)
-            .open(&fifo_path)
-            .map(|t| Writer {
-                line_writer: io::LineWriter::new(t),
-            })
-    }
-}
-
-impl io::Write for Writer {
-    fn write(&mut self, msg: &[u8]) -> Result<(usize)> {
-        self.line_writer.write_all(msg).map(|()| msg.len())
-    }
-
-    fn flush(&mut self) -> Result<()> {
-        self.line_writer.flush()
-    }
-}
+type FifoLineWriter = io::LineWriter<File>;
```

Comment:
How about `FcLineWriter` because it is a specialization of a `LineWriter` for the Firecracker usecase. Using `Fifo` in the name implies that it can only use fifo, but as you mentioned in your previous comment it works with regular files as well.

Reply:
Done. Also I renamed `open_fifo` to `open_fd_nonblock`.